### PR TITLE
fix: cancelAnimationFrame

### PR DESCRIPTION
**Pull request by GuoBinyong**
Fix logic errors in `AnimationLoop._cancelAnimationFrame`

### CODE REVIEW

File: modules/engine/src/animation-loop/animation-loop.ts [hunk 1]
```diff
@@ -304,7 +304,7 @@ export class AnimationLoop {
   }
 
   _cancelAnimationFrame() {
-    if (this._animationFrameId !== null) {
+    if (this._animationFrameId == null) {
```
Comment (ibgreen) [inline modules/engine/src/animation-loop/animation-loop.ts:307]: Nit: We use `===` for comparisons.

**Review by ibgreen — APPROVED** [review]

File: modules/engine/src/animation-loop/animation-loop.ts (hunk 1 above)
Comment (GuoBinyong) [reply modules/engine/src/animation-loop/animation-loop.ts:307]: @ibgreen  The reason for using '==' is that it can compare null and 'undefined'

Comment (GuoBinyong) [reply modules/engine/src/animation-loop/animation-loop.ts:307]: @ibgreen The reason for using '==' is that it can compare `null` and `undefined`

Comment (ibgreen) [reply modules/engine/src/animation-loop/animation-loop.ts:307]: Yes I understand, but for that case, we just use boolean negation. I find the difference between `===` and `==` too subtle and a cause of hard to spot errors, so we just avoid `==`. But that is of course just our preference in this particular code base.

**Pull request merged**